### PR TITLE
T14723 mbstring

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: bug fix
* Link to issue: #14723 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Changed methods internally to use `mbstring` equivalent ones (`strtolower`, etc.) making `mbstring` a prerequisite

Thanks

### CODE REVIEW

**Review by CameronHall — APPROVED** [review]

**Comment by sergeyklay** [comment]
Please add mbstring to the list of the required extension https://github.com/phalcon/cphalcon/blob/ef31ca8fdfdcbaeef24bd93a32107a43b6c1ba6b/config.json#L17-L26

File: phalcon/Config/ConfigFactory.zep [hunk 1]
```diff
@@ -97,11 +97,11 @@ class ConfigFactory extends AbstractFactory
             );
         }
 
-        let adapter = strtolower(config["adapter"]),
+        let adapter = mb_strtolower(config["adapter"]),
```
Comment (sergeyklay) [inline phalcon/Config/ConfigFactory.zep]: Is there any reason to use multi byte function for the adapter name? Do we plan provide support for PHP non in ASCII? AFAIK, mbstring functions are slower. So, it is curious to me, what we offer for such the price? Adapter names in national encodings?

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: phalcon/Config/ConfigFactory.zep (hunk 1 above)
Comment (sergeyklay) [reply phalcon/Config/ConfigFactory.zep]: I don't mind paying where it is necessary. But I don't quite understand why I should always pay for what I don't use.

**Comment by niden** [comment]
Closing this. We need to run more tests since this impacts performance especially in the HTTP layer

**Pull request closed without merging**